shared/editor/nodes/Heading.ts

```diff
@@ -274,7 +274,23 @@ export default class Heading extends Node {
       },
     });
 
-    return [new FoldingHeadersPlugin(this.editor.props.id), plugin];
+    const foldPlugin: Plugin = new Plugin({
+      props: {
+        decorations: (state) => {
+          const { doc } = state;
+          const decorations: Decoration[] = findCollapsedNodes(doc).map(
+            (block) =>
+              Decoration.node(block.pos, block.pos + block.node.nodeSize, {
+                class: "folded-content",
+              })
+          );
+
+          return DecorationSet.create(doc, decorations);
```

Comment:
Could be further improved to not create all decorations at once, but rather add/remove a particular decoration based on emitted transaction.

Reply:
Yea, really need to cache these in state